src/atlas/projection/detail/StretchRegular.cc

```diff
@@ -86,17 +87,82 @@ StretchLAM<Rotation>::StretchLAM(const eckit::Parametrisation& proj_st) :
+            ny_ = 0;
+
+            if (var_ratio_ == 1) {
+                   lam_hires_size = deltax_all;
+                   phi_hires_size = deltay_all;
+                   lambda_start = x_reg_start_;
+                   phi_start = y_reg_start_;
+
+                 } else {
+                      lam_hires_size = x_reg_end_ - x_reg_start_;
+                      phi_hires_size = y_reg_end_ - y_reg_start_;
+
+                      /** check for start of the new regular LAM grid x and y
+                      * in the middle of the previous regular grid
+                      */
+
+                      ///< distance end of the grid and internal regular grid
+                      add_xf_ = (deltax_all + epstest - lam_hires_size) / 2.;
+                      add_yf_ = (deltay_all + epstest - phi_hires_size) / 2.;
+                      /**
+                       *  Compute the number of points for different part of the grid
+                       *  internal regular grid high resolution
+                       *  stretched grid
+                       *  external regular grid low resolution
+                       *  +1 otherwise I have the intervals not the points
+                      */
+                      n_x_rim_ = rim_widthx_/delta_low_ ;
+                      n_y_rim_ = rim_widthy_/delta_low_ ;
+                      n_stretchedx_ = ((deltax_all + epstest - lam_hires_size) / delta_high_) - n_x_rim_;
+                      n_stretchedy_ = ((deltay_all + epstest - phi_hires_size) / delta_high_) - n_y_rim_ ;
+                      lambda_start = x_reg_start_;
+                      phi_start = y_reg_start_;
+                      /**
+                       *  check if stretched grid is in the middle
+                       *  of the previous regular grid
+                      */
+                      check_x = startx_ + add_xf_ - lambda_start;
+                      check_y = starty_ + add_yf_ - phi_start;
+                      check_st = n_stretchedx_ - n_stretchedy_;
+                      checkvalue(epsilon, check_x);
+                      checkvalue(epsilon, check_y);
+                      checkvalue(epsilon, check_st);
+
+                 }
+
+            nx_ = ((deltax_all + epstest) / delta_high_) + 1;
+            ny_ = ((deltay_all + epstest) / delta_high_ ) + 1;
+
+
+
 }
 
+
 template<typename Rotation>
 void StretchLAM<Rotation>::checkvalue(const double & epsilon,
                                         const double & value_check) const {
     std::string err_message;
     std::string str = std::to_string(value_check);
-    err_message = "USER defined limits not in the middle of the area " + str;
+    //err_message = "USER defined limits not in the middle of the area " + str;
     if (value_check > epsilon || value_check < (-1. * epsilon))
           {
-             throw eckit::BadValue(err_message, Here());
+             throw eckit::BadValue("USER defined limits not in the middle of the area " + str, Here());
```

Comment:
Already better, but you can also bring 
```c++
std::string err_message;
std::string str = std::to_string(value_check);
```
inside the `if` although now it is not as critical anymore since it is only done a few times in the constructor only rather than for each `xy2lonlat()`

Reply:
done